uk.co.compucorp.civicrm.civihrsampledata/civihrsampledata.php

```diff
@@ -130,7 +130,9 @@ function _civihrsampledata_importSampleData($extensionDirectory) {
     $fileToImport = new SplFileObject("{$csvDir}/{$file}.csv");
 
     $importerClassName = "CRM_CiviHRSampleData_Importer_{$class}";
-    $importer = new $importerClassName($fileToImport);
+    $importer = new $importerClassName();
+    $importer->setSplFileObject($fileToImport);
```

Comment:
passing the dependency via the constructor is what I tried to do at first , but doing this will require repeating passing the dependency for all child classes that implements its own constructor which I think is not Ideal .. so doing it via setter was a cleaner solution to this in my opinion .